src/builtin.c

```diff
@@ -363,6 +363,33 @@ static jv f_tonumber(jq_state *jq, jv input) {
   return type_error(input, "cannot be parsed as a number");
 }
 
+static jv f_toint(jq_state *jq, jv input) {
+  input = f_tonumber(jq, input);
+  if (!jv_is_valid(input))
+    return input;
+  if (jv_is_int64(input) || jv_is_uint64(input))
+    return input;
+  double d = jv_number_value(input);
+  if (d < 0 && d >= INT64_MIN) {
+    int64_t i = d;
+    if (i < 0)
+      return jv_int64(i);
+  } else if (d < UINT64_MAX) {
+    uint64_t u = d;
+    if (d == (double)u)
```

Comment:
Ditto.  Here we know that `d` is positive, and if it's also less than `UINT64_MAX` then we can represent it as a `uint64_t`.